### PR TITLE
[DOC] Cluster Operator deployment update

### PR DESCRIPTION
**Pull request by PaulRMellor**
### Type of change
- Documentation

### Description

Update to the overview of the Cluster Operator to clarify the deployment options.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by PaulRMellor** [comment]
I don't quite _get_ why the Cluster Operator sections are repeated. If we wanted to trim the doc a bit, we might keep the deployment procedures in 'Getting Started' and the description of the Cluster Operator in the later 'Cluster Operator' section.

File: documentation/book/assembly-cluster-operator.adoc [hunk 1]
```diff
@@ -8,16 +8,12 @@
...
 = Cluster Operator
 
-{ProductName} uses the Cluster Operator to deploy and manage Kafka (including Zookeeper) and Kafka Connect clusters.
-The Cluster Operator is deployed inside of the Kubernetes cluster.
-To deploy a Kafka cluster, a `Kafka` resource with the cluster configuration has to be created within the Kubernetes cluster.
-Based on what is declared inside of the `Kafka` resource, the Cluster Operator deploys a corresponding Kafka cluster.
-For more information about the different configuration options supported by the `Kafka` resource, see xref:assembly-deployment-configuration-kafka-str[]
-
-NOTE: {ProductName} contains example YAML files, which make deploying a Cluster Operator easier.
+{ProductName} uses the Cluster Operator to deploy and manage Kafka (including Zookeeper) and Kafka Connect clusters in a Kubernetes cluster.
```
Comment (ppatierno) [inline documentation/book/assembly-cluster-operator.adoc]: it's not true anymore we should mention Kafka Mirror Maker and the Kafka Bridge as well.

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* xref:deploying-cluster-operator-to-watch-whole-cluster-{context}[All namespaces]
+
+NOTE: {ProductName} provides example YAML files to make the deployment process easier.
+
+Cluster Operator watches the following resources:
+
+* A `Kafka` resource for the Kafka cluster.
+* A `KafkaConnect` resource for the Kafka Connect cluster.
+* A `KafkaConnectS2I` resource for the Kafka Connect cluster with Source2Image support.
+* A `KafkaMirrorMaker` resource for the Kafka Mirror Maker instance.
```
Comment (ppatierno) [inline documentation/book/con-cluster-operator-watch-options.adoc]: We should add "A `KafkaBridge` resource for the Kafka Bridge instance"

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+When a new `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, or `Kafka Mirror Maker` resource is created in the Kubernetes cluster, the operator gets the cluster description from the resource and starts creating a new Kafka, Kafka Connect, or Kafka Mirror Maker cluster by creating the necessary Kubernetes resources, such as StatefulSets, Services and ConfigMaps.
```
Comment (ppatierno) [inline documentation/book/con-cluster-operator-watch-options.adoc]: Even here `KafkaBridge` has to be added everywhere.

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Each time a Kafka resource is updated, the operator performs corresponding updates on the Kubernetes resources that make up the Kafka, Kafka Connect, or Kafka Mirror Maker cluster.
```
Comment (ppatierno) [inline documentation/book/con-cluster-operator-watch-options.adoc]: Here again, Kafka Bridge.

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Resources are either patched or deleted, and then recreated in order to make the Kafka, Kafka Connect, or Kafka Mirror Maker cluster reflect the desired state of the cluster resource. This operation might cause a rolling update which might lead to service disruption.
```
Comment (ppatierno) [inline documentation/book/con-cluster-operator-watch-options.adoc]: One more time Kafka Bridge.

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 6]
```diff
@@ -1,34 +1,27 @@
...
+The Cluster Operator can also deploy:
 
-When a new `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, or `Kafka Mirror Maker` resource is created in the Kubernetes cluster, the operator gets the cluster description from the desired resource and starts creating a new Kafka, Kafka Connect, or Kafka Mirror Maker cluster by creating the necessary other Kubernetes resources, such as StatefulSets, Services, ConfigMaps, and so on.
+* A Topic Operator to provide operator-style topic management via `KafkaTopic` custom resources.
+* A Kafka Connect cluster to connect to an existing Kafka cluster.
 
-Every time the desired resource is updated by the user, the operator performs corresponding updates on the Kubernetes resources which make up the Kafka, Kafka Connect, or Kafka Mirror Maker cluster.
-Resources are either patched or deleted and then re-created in order to make the Kafka, Kafka Connect, or Kafka Mirror Maker cluster reflect the state of the desired cluster resource.
-This might cause a rolling update which might lead to service disruption.
+NOTE: On OpenShift, a Kafka Connect deployment can incorporate a Source2Image feature to provides a convenient way to include connectors.
```
Comment (ppatierno) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: Here are missing both Kafka Mirror Maker and Kafka Bridge.

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+// assembly-operators-cluster-operator.adoc
+// assembly-cluster-operator.adoc
+
+[id='con-cluster-operator-watch-options-{context}']
+
+= Watch options for a Cluster Operator deployment
+
+When the Cluster Operator is running, it starts to _watch_ Kafka resources for updates.
+
+Depending on the deployment, the Cluster Operator can _watch_ Kafka resources from:
```
Comment (laidan6000) [inline documentation/book/con-cluster-operator-watch-options.adoc]:
```suggestion
Depending on the deployment, the Cluster Operator can watch Kafka resources from:
```

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+* xref:deploying-cluster-operator-{context}[A single namespace (the namespace it is installed)]
+* xref:deploying-cluster-operator-to-watch-multiple-namespaces{context}[Multiple namespaces]
+* xref:deploying-cluster-operator-to-watch-whole-cluster-{context}[All namespaces]
+
+NOTE: {ProductName} provides example YAML files to make the deployment process easier.
+
+Cluster Operator watches the following resources:
```
Comment (laidan6000) [inline documentation/book/con-cluster-operator-watch-options.adoc]:
```suggestion
The Cluster Operator watches the following resources:
```

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 3 above)
Comment (laidan6000) [inline documentation/book/con-cluster-operator-watch-options.adoc]: Remove spaces: `Kafka Mirror Maker` resource is created

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 5 above)
Comment (laidan6000) [inline documentation/book/con-cluster-operator-watch-options.adoc]:
```suggestion
Resources are either patched or deleted, and then recreated in order to make the Kafka, Kafka Connect, or Kafka Mirror Maker cluster reflect the desired state of the cluster resource. This operation might cause a rolling update that might lead to service disruption.
```

File: documentation/book/assembly-cluster-operator.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/book/assembly-cluster-operator.adoc]: Thanks. I've updated as a list. 

{ProductName} uses the Cluster Operator to deploy to a Kubernetes cluster and manage the following:

* Kafka (including Zookeeper)
* Kafka Connect
* Kafka Mirror Maker
* Kafka Bridge

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/book/con-cluster-operator-watch-options.adoc]: I think it's easier to reduce this to: When one of these resources is created in...

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 5 above)
Comment (PaulRMellor) [reply documentation/book/con-cluster-operator-watch-options.adoc]: I've rewritten this section to be more generic about clusters/resources

File: documentation/book/con-what-the-cluster-operator-does.adoc (hunk 6 above)
Comment (PaulRMellor) [reply documentation/book/con-what-the-cluster-operator-does.adoc]: Updated, as with the other intro

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/book/con-cluster-operator-watch-options.adoc]: This section was rewritten, so change not required.

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 4 above)
Comment (PaulRMellor) [reply documentation/book/con-cluster-operator-watch-options.adoc]: All updated in a generic way

File: documentation/book/con-cluster-operator-watch-options.adoc [hunk 9]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// assembly-operators-cluster-operator.adoc
+// assembly-cluster-operator.adoc
+
+[id='con-cluster-operator-watch-options-{context}']
+
+= Watch options for a Cluster Operator deployment
+
+When the Cluster Operator is running, it starts to _watch_ Kafka resources for updates.
```
Comment (scholzj) [inline documentation/book/con-cluster-operator-watch-options.adoc]: Would this be better:

```suggestion
When the Cluster Operator is running, it starts to _watch_ for Kafka resources and their updates.
```

File: documentation/book/assembly-cluster-operator.adoc [hunk 10]
```diff
@@ -8,16 +8,17 @@
 [id='cluster-operator-{context}']
 = Cluster Operator
 
-{ProductName} uses the Cluster Operator to deploy and manage Kafka (including Zookeeper) and Kafka Connect clusters.
-The Cluster Operator is deployed inside of the Kubernetes cluster.
-To deploy a Kafka cluster, a `Kafka` resource with the cluster configuration has to be created within the Kubernetes cluster.
-Based on what is declared inside of the `Kafka` resource, the Cluster Operator deploys a corresponding Kafka cluster.
-For more information about the different configuration options supported by the `Kafka` resource, see xref:assembly-deployment-configuration-kafka-str[]
+{ProductName} uses the Cluster Operator to deploy to a Kubernetes cluster and manage the following:
```
Comment (scholzj) [inline documentation/book/assembly-cluster-operator.adoc]: This sounds a bit strange to me. I had to read it twice to understand what it is trying to say. I wonder if we can somehow rephrase it to make it more understandable.

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 11]
```diff
@@ -1,34 +1,31 @@
...
+The Cluster Operator can deploy clusters for:
 
-.Example architecture for the Cluster Operator
+* Kafka (including Zookeeper)
+* Kafka Connect
+* Kafka Mirror Maker
+* Kafka Bridge
 
-image:cluster-operator.png[Cluster Operator]
+The Cluster Operator can also deploy (through Entity Operator configuration)
```
Comment (scholzj) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: Should we make it clear that this is only with the `Kafka` resource?

File: documentation/book/con-cluster-operator-watch-options.adoc (hunk 9 above)
Comment (PaulRMellor) [reply documentation/book/con-cluster-operator-watch-options.adoc]: Changed to 'watch for updates of Kafka resources'

File: documentation/book/assembly-cluster-operator.adoc (hunk 10 above)
Comment (PaulRMellor) [reply documentation/book/assembly-cluster-operator.adoc]: I've rewritten. I've also moved, as we were saying pretty much the same in the next section. Each operator assembly now has a single line to say what it does. The following concepts give more detail.

File: documentation/book/con-what-the-cluster-operator-does.adoc (hunk 11 above)
Comment (PaulRMellor) [reply documentation/book/con-what-the-cluster-operator-does.adoc]: Definitely. I've reworded and rearranged to make more clear.

**Comment by PaulRMellor** [comment]
@scholzj - Ready to merge
